aten/src/ATen/core/ATenDispatch.h

```diff
@@ -15,20 +18,34 @@
 
 namespace at {
 
+template <typename R, typename... Args>
+struct lazy;
+template <typename R, typename... Args>
+typename lazy<R, Args...>::FuncType make_lazy(
+    R (*fn)(Args...),
+    const char* schema);
+
 // ATenOpTable stores the implementations for each backend, in addition to
 // an implementation for variables.
 class CAFFE2_API ATenOpTable {
  public:
   ATenOpTable(std::string schema)
     : schema_(std::move(schema)) {}
 
-  template<class FuncType>
-  FuncType* getOp(Backend backend, bool is_variable) const {
+  template <class FuncType>
+  std::function<FuncType> getOp(Backend backend, bool is_variable) const {
+    if (schema_ == "aten::to_eager(Tensor self) -> Tensor") {
+      return reinterpret_cast<FuncType*>(getBaseOp(Backend::CPU));
+    }
+    if (backend == Backend::Lazy) {
+      return make_lazy((FuncType*)nullptr, schema_.c_str());
+    }
     if (is_variable) {
       return reinterpret_cast<FuncType*>(getVariableOp());
     }
     return reinterpret_cast<FuncType*>(getBaseOp(backend));
   }
```

Comment:
Repeating from what was discussed in realtime: these changes will conflict with the post-multi dispatch implementation at  https://github.com/pytorch/pytorch/blob/7bfa1e4fa1bb9823d9bf6f706dcfa8e1380905cc/aten/src/ATen/core/ATenDispatch.h#L63

Reply:
The recommendation:

Currently, the implementation of dispatcher logic in https://github.com/pytorch/pytorch/pull/25653

```
op = dispatch_table[op_id][valid_keys[0]]
if not op:
  # fallback logic...
return op(args)
```

We adjust fallback handling to now be:

```
op = dispatch_table[op_id][valid_keys[0]]
if not op and default_dispatch_table[valid_keys[0]]:
  op = BoxedOpFunctor<FuncType>(default_dispatch_table[valid_keys[0]])
if not op:
  # fallback logic...
return op(args)
```

There are two pieces:

1. The default dispatch table contains registrations of functions default_dispatch(Operator op, Stack&& ivalues) which get passed in what operator they need to process, and arguments in boxed form. The expectation is a backend like lazy would write this function once to work polymorphically over everything. Operator is some type that gives you the schema string, and also a way to redispatch on the operator.
2. The boxed op functors boxed_op_functors is a functor which unpacks FuncType into its boxed form, and then calls the default dispatch function. It is templated here, as is done in this PR.